frontend/packages/topology/src/elements/BaseNode.ts

```diff
@@ -133,6 +143,21 @@ export default class BaseNode<E extends NodeModel = NodeModel, D = any> extends
     return this.group;
   }
 
+  isCollapsed(): boolean {
+    return this.collapsed;
+  }
+
+  setCollapsed(collapsed: boolean): void {
+    if (collapsed !== this.collapsed) {
+      const prevCenter = this.getBounds()
+        .getCenter()
+        .clone();
+      this.collapsed = collapsed;
+      this.getBounds().setCenter(prevCenter.x, prevCenter.y);
```

Comment:
We should have a TODO with a comment why this is here so we can address it better.

Reply:
added